pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -79,33 +80,46 @@ type ecrProvider struct {
 	region    string
 	regionURL string
 	getter    tokenGetter
+	account   string
 }
 
 var _ credentialprovider.DockerConfigProvider = &ecrProvider{}
 
 // registryURL has different suffix in AWS China region
-func registryURL(region string) string {
+func registryURL(account string, region string) string {
 	dnsSuffix := awsStandardDNSSuffix
 	// deal with aws none standard regions
 	if strings.HasPrefix(region, awsChinaRegionPrefix) {
 		dnsSuffix = awsChinaDNSSuffix
 	}
-	return fmt.Sprintf(registryURLTemplate, region, dnsSuffix)
+	return fmt.Sprintf(registryURLTemplate, account, region, dnsSuffix)
 }
 
-// RegisterCredentialsProvider registers a credential provider for the specified region.
-// It creates a lazy provider for each AWS region, in order to support
-// cross-region ECR access. They have to be lazy because it's unlikely, but not
-// impossible, that we'll use more than one.
+// RegisterCredentialsProviders registers the credential providers for the appropriate regions and accounts.
 // This should be called only if using the AWS cloud provider.
 // This way, we avoid timeouts waiting for a non-existent provider.
-func RegisterCredentialsProvider(region string) {
-	glog.V(4).Infof("registering credentials provider for AWS region %q", region)
+func RegisterCredentialsProviders(regions []string, accounts []string) {
+	if regions == nil {
+		return
+	}
+	if accounts == nil || len(accounts) == 0 {
+		accounts = []string{"*"}
+	}
+	for _, region := range regions {
+		for _, account := range accounts {
+			registerCredentialsProvider(region, account)
+		}
+	}
+}
 
-	credentialprovider.RegisterCredentialProvider("aws-ecr-"+region,
+func registerCredentialsProvider(region string, account string) {
+	glog.V(4).Infof("registering credentials provider for AWS regions %q", region)
+
+	credentialprovider.RegisterCredentialProvider("aws-ecr-"+region+"-"+account,
```

Comment:
Just an observation: We've technically changed the name here for the "*" case, I don't _think_ it matter though.

Reply:
I'm not sure what you mean. Changed the name of the credential provider?